sycl/tools/lspi.cpp

```diff
@@ -0,0 +1,167 @@
+// with the more complex "check-sycl" utility.
+//
+// There are two types of output:
+//   concise (default) and
+//   verbose (enabled with any argument).
+//
+// In verbose mode it also prints, which devices would be chose  by various SYCL
+// device selectors.
+//
+#include <CL/sycl.hpp>
+
+#include <cstdlib>
+#include <iostream>
+#include <stdlib.h>
+
+using namespace cl::sycl;
+using namespace std;
+
+// Controls verbose output vs. concise.
+bool verbose = false;
+
+// Trivial custom selector that selects a device of the given type.
+class custom_selector : public device_selector {
+  info::device_type MType;
+
+public:
+  custom_selector(info::device_type Type) : MType(Type) {}
+  int operator()(const device &Dev) const override {
+    return Dev.get_info<info::device::device_type>() == MType ? 1 : -1;
+  }
+};
+
+static void printDeviceInfo(const device &Device, const string &Prepend) {
+  auto DeviceType = Device.get_info<info::device::device_type>();
+  string DeviceTypeName;
+  switch (DeviceType) {
+  case info::device_type::cpu:
+    DeviceTypeName = "CPU ";
+    break;
+  case info::device_type::gpu:
+    DeviceTypeName = "GPU ";
+    break;
+  case info::device_type::host:
+    DeviceTypeName = "HOST";
+    break;
+  case info::device_type::accelerator:
+    DeviceTypeName = "ACC ";
+    break;
+  default:
+    DeviceTypeName = "??? ";
+    break;
+  }
+
+  auto DeviceVersion = Device.get_info<info::device::version>();
+  auto DeviceName = Device.get_info<info::device::name>();
+  auto DeviceVendor = Device.get_info<info::device::vendor>();
+  auto DeviceDriverVersion = Device.get_info<info::device::driver_version>();
+
+  if (verbose) {
+    cout << Prepend << "Type       : " << DeviceTypeName << std::endl;
+    cout << Prepend << "Version    : " << DeviceVersion << std::endl;
+    cout << Prepend << "Name       : " << DeviceName << std::endl;
+    cout << Prepend << "Vendor     : " << DeviceVendor << std::endl;
+    cout << Prepend << "Driver     : " << DeviceDriverVersion << std::endl;
+  } else {
+    cout << Prepend << DeviceTypeName << ": " << DeviceVersion << "[ "
+         << DeviceDriverVersion << " ]" << std::endl;
+  }
+}
+
+static void printSelectorChoice(const device_selector &Selector,
+                                const std::string &Prepend) {
+  try {
+    const auto &Dev = device(Selector);
+    printDeviceInfo(Dev, Prepend);
+
+  } catch (const cl::sycl::runtime_error &Exception) {
+    // Truncate long string so it can fit in one-line
+    std::string What = Exception.what();
+    if (What.length() > 50)
+      What = What.substr(0, 50) + "...";
+
+    if (verbose) {
+      cout << Prepend << "Type       : " << What << std::endl;
+      cout << Prepend << "Version    : " << What << std::endl;
+      cout << Prepend << "Name       : " << What << std::endl;
+      cout << Prepend << "Vendor     : " << What << std::endl;
+      cout << Prepend << "Driver     : " << What << std::endl;
```

Comment:
Could you clarify what this code is doing?
Is it trying to mimic the format of "successful print"?
Why?

Reply:
Yes, same format as in successful output. It was useful to "grep" the output in the LIT. Of course, I can change this if you feel strongly.

Reply:
But I no longer use it now, so removed.